quickwit-metastore/src/metastore/file_backed_metastore/index.rs

```diff
@@ -29,28 +30,83 @@ use crate::{IndexMetadata, MetastoreError, MetastoreResult, Split, SplitMetadata
 
 /// A `Index` object carries an index metadata and its split metadata.
 #[derive(Clone, Debug, Serialize, Deserialize)]
-pub(crate) struct Index {
+#[serde(into = "VersionedIndex")]
+#[serde(from = "VersionedIndex")]
+pub struct Index {
     /// Metadata specific to the index.
-    index: IndexMetadata,
+    pub metadata: IndexMetadata,
```

Comment:
Was there a particular reason for keeping the fields `index` and `splits` private? I can also leave them private and write a constructor.

Reply:
I renamed `index` to `metadata` because `index.index` was awkward. The serialized version is still called `index`.

Reply:
> I renamed index to metadata because index.index was awkward. The serialized version is still called index.

It was awkward. But now the discrepancy between serialization and the object is awkward. That's a pick your poison story.

Reply:
> Was there a particular reason for keeping the fields index and splits private? I can also leave them private and write a constructor.

I suspect the goal is to ensure people go through the methods to change index and splits?
It enforces a bunch of cool stuff on the state.